### PR TITLE
Added temporary workaround for sharing paused time

### PR DESCRIPTION
**Pull request by BanCrash**
I have verified that:
- [x] These changes work, by building the extension and testing.
- [x] That the changes comply with the [rules](https://github.com/darklinkpower/PlayniteExtensionsCollection#contributing) indicated in the repository.
- [x] Pull request is targeting `master` branch.

This PR is linked to this PR of GameActivity extension: https://github.com/Lacro59/playnite-gameactivity-plugin/pull/119

Temporary workaround for #122

I used the simple code as possible since this is only a temporary workaround until Playnite allows to share data among extensions.

The file is placed inside AppData/Roaming/Playnite/ExtensionsData folder with the name of "PlayState.txt". This file will be cleaned after executing one game or retrieving the info on GameActivity plugin in order to avoid reusing the same information over and over again if PlayState gets uninstalled or is not working for any reason.

I added as much code as possible in new methods, notes for explaining the functioning and a annotation that says:

> // Temporary workaround for sharing PlayState paused time until Playnite allows to share data among extensions

In order to ease the removal in the future when another better method is available.

EDIT: I forgot to mention, I tested the extension manually replacing the compiled files, I didn't achieve to create the pext file.

### CODE REVIEW

File: source/Generic/PlayState/PlayState.cs [hunk 1]
```diff
@@ -591,6 +594,22 @@ private void RemoveGameStopwatchTuple(Game game)
             }
         }
 
+        private void InitializePlaytimeInfoFile() // Temporary workaround for sharing PlayState paused time until Playnite allows to share data among extensions
+        {
+            // This method will remove the info of the txt file in order to avoid reusing the previous play information.
+            string[] info = { " ", " " };
+
+            File.WriteAllLines(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Playnite", "ExtensionsData", "PlayState.txt"), info);
```
Comment (darklinkpower) [inline source/Generic/PlayState/PlayState.cs:602]: Please use the `ExtensionsDataPath` path from the Paths API and not environment variables https://playnite.link/docs/api/Playnite.SDK.IPlaynitePathsAPI.html#Playnite_SDK_IPlaynitePathsAPI_ExtensionsDataPath

Should be `PlayniteApi.Paths.ExtensionsDataPath`. You can use the `ConfigurationPath` if you prefer also but preferably `ExtensionsDataPath`

File: source/Generic/PlayState/PlayState.cs [hunk 2]
```diff
@@ -591,6 +594,22 @@ private void RemoveGameStopwatchTuple(Game game)
...
+        }
+
+        private void ExportPausedTimeInfo(Game game, ulong elapsedSeconds) // Temporary workaround for sharing PlayState paused time until Playnite allows to share data among extensions
+        {
+            // This method will write the Id and pausedTime to PlayState.txt file placed inside ExtensionsData Roaming Playnite folder
+            string[] info = { game.Id.ToString(), elapsedSeconds.ToString() };
+
+            File.WriteAllLines(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Playnite", "ExtensionsData", "PlayState.txt"), info);
```
Comment (darklinkpower) [inline source/Generic/PlayState/PlayState.cs:610]: Same here

**Review by darklinkpower — CHANGES_REQUESTED** [review]

**Comment by darklinkpower** [comment]
Another thing. Is the order of execution in the `OnGameStopped` PlayState -> GameActivity? Because if it's the other way around, then this will not work.

File: source/Generic/PlayState/PlayState.cs (hunk 1 above)
Comment (BanCrash) [reply source/Generic/PlayState/PlayState.cs:602]: I will check it, thanks!

**Comment by BanCrash** [comment]
@darklinkpower done!

**Comment by darklinkpower** [comment]
Be aware that `ExtensionsDataPath` is plugin specific so you will need to combine paths to get the location of this plugin's `ExtensionsDataPath` from another plugin.  `ConfigurationPath` is global for all plugins so it's also ok if you want to use that instead.

**Comment by BanCrash** [comment]
> Be aware that `ExtensionsDataPath` is plugin specific so you will need to combine paths to get the location of this plugin's `ExtensionsDataPath` from another plugin. `ConfigurationPath` is global for all plugins so it's also ok if you want to use that instead.

I already tested with the last version of this PR and the last version of GameActivity PR and it's working as before changing the path. GameActivity is reading the file and substracting the paused time from its data as expected:

image
(After few seconds I paused for 1 minute and this is the result).

Also, I modified now the PlayState to log the path to assure it and this is the result:
image

> 01-03 23:28:38.147|DEBUG|PlayState#PlayState:C:\Users\User\AppData\Roaming\Playnite\ExtensionsData\PlayState.txt

**Comment by darklinkpower** [comment]
Oh you are right, I was confusing it with this, my bad https://playnite.link/docs/api/Playnite.SDK.Plugins.Plugin.html#Playnite_SDK_Plugins_Plugin_GetPluginUserDataPath

**Pull request merged**